superset-frontend/src/middleware/asyncEvent.ts

```diff
@@ -25,178 +24,240 @@ import {
   parseErrorJson,
 } from '../utils/getClientErrorObject';
 
-export type AsyncEvent = {
-  id: string;
+type AsyncEvent = {
+  id?: string | null;
   channel_id: string;
   job_id: string;
-  user_id: string;
+  user_id?: string;
   status: string;
-  errors: SupersetError[];
-  result_url: string;
-};
-
-type AsyncEventOptions = {
-  config: {
-    GLOBAL_ASYNC_QUERIES_TRANSPORT: string;
-    GLOBAL_ASYNC_QUERIES_POLLING_DELAY: number;
-  };
-  getPendingComponents: (state: any) => any[];
-  successAction: (componentId: number, componentData: any) => { type: string };
-  errorAction: (componentId: number, response: any) => { type: string };
-  processEventsCallback?: (events: AsyncEvent[]) => void; // this is currently used only for tests
+  errors?: SupersetError[];
+  result_url: string | null;
 };
 
 type CachedDataResponse = {
-  componentId: number;
   status: string;
   data: any;
 };
+type AppConfig = Record<string, any>;
+type ListenerFn = (asyncEvent: AsyncEvent) => Promise<any>;
```

Comment:
Bycatch: I believe the correct type for `data` should be `LegacyQueryData | ChartDataResponseResult[]`

Reply:
This could be a response payload for SQL Lab as well (in the future), which is the reason why it's generic. Well, that and laziness.

Reply:
Oh right, that makes total sense 👍